Cannot load a saved GCN_LSTM model: TypeError: __init__() missing 1 required positional argument: 'A'

### Describe the bug

An `GCN_LSTM` model cannot be loaded, because it hits an error in `FixedAdjacencyGraphConvolution.__init__`.


https://buildkite.com/stellar/stellargraph-public/builds/4753#e083dc95-02d2-40b4-95b1-81d0770d383a/322-984
### To Reproduce

Steps to reproduce the behavior:

1. Run the tests added in #1676

### Observed behavior

```
_____________________ test_gcn_lstm_save_load[univariate] ______________________
 
tmpdir = local('/tmp/pytest-of-root/pytest-0/test_gcn_lstm_save_load_univar0')
arange_graph = <stellargraph.core.graph.StellarGraph object at 0x7f473ecf24e0>
 
    def test_gcn_lstm_save_load(tmpdir, arange_graph):
        gen = SlidingFeaturesNodeGenerator(arange_graph, 2, batch_size=3)
        gcn_lstm = GCN_LSTM(None, None, [2], [4], generator=gen)
>       test_utils.model_save_load(tmpdir, gcn_lstm)
 
tests/layer/test_gcn_lstm.py:223:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_utils/__init__.py:39: in model_save_load
    loaded = keras.models.load_model(str(saved_dir), sg.custom_keras_layers)
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/save.py:190: in load_model
    return saved_model_load.load(filepath, compile)
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:116: in load
    model = tf_load.load_internal(path, loader_cls=KerasObjectLoader)
/usr/local/lib/python3.6/site-packages/tensorflow/python/saved_model/load.py:604: in load_internal
    export_dir)
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:188: in __init__
    super(KerasObjectLoader, self).__init__(*args, **kwargs)
/usr/local/lib/python3.6/site-packages/tensorflow/python/saved_model/load.py:123: in __init__
    self._load_all()
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:209: in _load_all
    self._layer_nodes = self._load_layers()
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:309: in _load_layers
    layers[node_id] = self._load_layer(proto.user_object, node_id)
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:335: in _load_layer
    obj, setter = self._revive_from_config(proto.identifier, metadata, node_id)
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:353: in _revive_from_config
    self._revive_layer_from_config(metadata, node_id))
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/saving/saved_model/load.py:408: in _revive_layer_from_config
    generic_utils.serialize_keras_class_and_config(class_name, config))
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/layers/serialization.py:109: in deserialize
    printable_module_name='layer')
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py:375: in deserialize_keras_object
    return cls.from_config(cls_config)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 
cls = <class 'stellargraph.layer.gcn_lstm.FixedAdjacencyGraphConvolution'>
config = {'activation': 'relu', 'bias_constraint': None, 'bias_initializer': None, 'bias_regularizer': None, ...}
 
    @classmethod
    def from_config(cls, config):
      """Creates a layer from its config.
    
      This method is the reverse of `get_config`,
      capable of instantiating the same layer from the config
      dictionary. It does not handle layer connectivity
      (handled by Network), nor weights (handled by `set_weights`).
    
      Arguments:
          config: A Python dictionary, typically the
              output of get_config.
    
      Returns:
          A layer instance.
      """
>     return cls(**config)
E     TypeError: __init__() missing 1 required positional argument: 'A'
 
/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py:655: TypeError
```
### Expected behavior

Loading should work.

### Environment

StellarGraph develop 8a3a74c


### Additional context

N/A

--- a/stellargraph/layer/gcn_lstm.py
+++ b/stellargraph/layer/gcn_lstm.py
@@ -137,16 +137,16 @@ class FixedAdjacencyGraphConvolution(Layer):
         """
         _batch_dim, n_nodes, features = input_shapes
 
-        if self.adj is None:
-            raise ValueError(
-                "A: expected adjacency matrix when initially defining layer, found None which is only supported when loading a saved model"
-            )
+        if self.adj is not None:
+            adj_init = initializers.constant(self.adj)
+        else:
+            adj_init = initializers.zeros()
 
         self.A = self.add_weight(
             name="A",
             shape=(n_nodes, n_nodes),
             trainable=False,
-            initializer=initializers.constant(self.adj),
+            initializer=adj_init,
         )
         self.kernel = self.add_weight(
             shape=(features, self.units),

--- a/stellargraph/layer/gcn_lstm.py
+++ b/stellargraph/layer/gcn_lstm.py
@@ -143,10 +143,7 @@ class FixedAdjacencyGraphConvolution(Layer):
             adj_init = initializers.zeros()
 
         self.A = self.add_weight(
-            name="A",
-            shape=(n_nodes, n_nodes),
-            trainable=False,
-            initializer=adj_init,
+            name="A", shape=(n_nodes, n_nodes), trainable=False, initializer=adj_init
         )
         self.kernel = self.add_weight(
             shape=(features, self.units),

--- a/stellargraph/layer/gcn_lstm.py
+++ b/stellargraph/layer/gcn_lstm.py
@@ -69,7 +69,8 @@ class FixedAdjacencyGraphConvolution(Layer):
             kwargs["input_shape"] = (input_dim,)
 
         self.units = units
-        self.adj = calculate_laplacian(A)
+        self.adj = calculate_laplacian(A) if A is not None else None
+
         self.activation = activations.get(activation)
         self.use_bias = use_bias
 
@@ -101,6 +102,10 @@ class FixedAdjacencyGraphConvolution(Layer):
             "bias_initializer": initializers.serialize(self.bias_initializer),
             "bias_regularizer": regularizers.serialize(self.bias_regularizer),
             "bias_constraint": constraints.serialize(self.bias_constraint),
+            # the adjacency matrix argument is required, but
+            # (semi-secretly) supports None for loading from a saved
+            # model, where the adjacency matrix is a saved weight
+            "A": None,
         }
 
         base_config = super().get_config()
@@ -131,6 +136,11 @@ class FixedAdjacencyGraphConvolution(Layer):
 
         """
         _batch_dim, n_nodes, features = input_shapes
+
+        if self.adj is None:
+            raise ValueError(
+                "A: expected adjacency matrix when initially defining layer, found None which is only supported when loading a saved model"
+            )
 
         self.A = self.add_weight(
             name="A",

--- a/tests/layer/test_gcn_lstm.py
+++ b/tests/layer/test_gcn_lstm.py
@@ -217,7 +217,6 @@ def test_gcn_lstm_generator(arange_graph):
     np.testing.assert_array_equal(predictions, predictions2)
 
 
-@pytest.mark.xfail(reason="FIXME #1681")
 def test_gcn_lstm_save_load(tmpdir, arange_graph):
     gen = SlidingFeaturesNodeGenerator(arange_graph, 2, batch_size=3)
     gcn_lstm = GCN_LSTM(None, None, [2], [4], generator=gen)

